src/Sylius/Bundle/ApiBundle/EventSubscriber/ChannelDeletionEventSubscriber.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\EventSubscriber;
+
+use ApiPlatform\Core\EventListener\EventPriorities;
+use Sylius\Bundle\ApiBundle\Exception\ChannelCannotBeRemoved;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/** @experimental */
+final class ChannelDeletionEventSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @param ChannelRepositoryInterface<ChannelInterface> $channelRepository
+     */
+    public function __construct(private ChannelRepositoryInterface $channelRepository)
+    {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::VIEW => [
+                ['protectFromRemovingTheOnlyChannelInStore', EventPriorities::PRE_WRITE],
+            ],
+        ];
+    }
+
+    public function protectFromRemovingTheOnlyChannelInStore(ViewEvent $event): void
+    {
+        $channel = $event->getControllerResult();
+        $method = $event->getRequest()->getMethod();
+
+        if (!$channel instanceof ChannelInterface || $method !== Request::METHOD_DELETE) {
+            return;
+        }
+
+        $results = $this->channelRepository->findBy(['enabled' => true]);
+
+        if (!$results || (count($results) === 1 && current($results) === $channel)) {
+            throw new ChannelCannotBeRemoved('The channel cannot be deleted. At least one enabled channel is required.');
```

Comment:
Would be good to sync terminology.

```suggestion
            throw new ChannelCannotBeRemoved('The channel cannot be removed. At least one enabled channel is required.');
```

Reply:
I'm not really convinced to make this change here, as it should be also adjusted to other exceptions that have just the same terminology, so maybe in another PR 🤔

Reply:
For me it isn't a good practice to postpone adjusting some things because other places are also violating some decisions. Especially so small thingies.

Reply:
It is only a message, and it is not consistent with the name of an exception. I would be for changing this small thing here 🖖🏻